de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/session/JoinSessionRequestHandler.java

```diff
@@ -0,0 +1,61 @@
+package de.fu_berlin.inf.dpp.server.session;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.jivesoftware.smack.PacketListener;
+import org.jivesoftware.smack.packet.Packet;
+
+import de.fu_berlin.inf.dpp.communication.extensions.JoinSessionRequestExtension;
+import de.fu_berlin.inf.dpp.net.IReceiver;
+import de.fu_berlin.inf.dpp.net.xmpp.JID;
+import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
+import de.fu_berlin.inf.dpp.util.NamedThreadFactory;
+
+/**
+ * Listens for and handles JoinSessionRequests allowing clients to join server sessions.
+ */
+public final class JoinSessionRequestHandler {
+    private final ISarosSessionManager sessionManager;
+
+    /**
+     * Invitation Requests are handled very fast and are almost immediately dispatched
+     * into another executor in the {@link NegotiationHandler}. More then one thread is currently
+     * not necessary and would likely just introduce more overhead.
+     * We just want asynchronous execution and a queue.
+     */
+    private final ExecutorService executor = Executors.newSingleThreadExecutor(
+        new NamedThreadFactory("JoinSessionRequestHandler")
+    );
+
+    private final PacketListener joinSessionRequestListener = new PacketListener() {
+        @Override
+        public void processPacket(final Packet packet) {
+            executor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    handleInvitationRequest(new JID(packet.getFrom()),
+                       JoinSessionRequestExtension.PROVIDER.getPayload(packet));
+                }
+            });
+        }
+    };
+
+    public JoinSessionRequestHandler(ISarosSessionManager sessionManager,
+        IReceiver receiver) {
+
+        this.sessionManager = sessionManager;
+        listenForJoinRequests(receiver);
+    }
+
+    private void listenForJoinRequests(IReceiver receiver) {
+        receiver.addPacketListener(joinSessionRequestListener,
+            JoinSessionRequestExtension.PROVIDER.getPacketFilter());
+    }
+
+    private void handleInvitationRequest(final JID from,
+        JoinSessionRequestExtension extension) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused method parameter "extension". ([squid:S1172](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=290&issues=AWbKHVqDLpqqSUlcAilo&open=AWbKHVqDLpqqSUlcAilo)

[ISSUE_KEY]: AWbKHVqDLpqqSUlcAilo